app/src/main/java/com/github/featuredetectandroid/utils/KeypointExtractionAlgorithm.kt

```diff
@@ -0,0 +1,31 @@
+package com.github.featuredetectandroid.utils
+
+import android.util.Log
+
+private const val TAG = "KeypointExtractionAlgorithm"
+
+enum class KeypointExtractionAlgorithm {
+    NONE,
+    SIFT,
+    SURF,
+    ORB,
+    SUPERPOINT
+}
+
+fun algorithmsNamesAsList() = enumValues<KeypointExtractionAlgorithm>().map {
```

Comment:
1. It can be a simple `val`, no need to recalculate this each time.
2. Let's name it `names` and move it into the `companion object` of `KeypointExtractionAlgorithm` -- this would be more readable.
3. After applying the changes above, you can replace `enumValues<KeypointExtractionAlgorithm>()` with just `values()`. And also simplify the calls to `NONE` and `SUPERPOINT` below.

Reply:
fixed